internal/dataplane/parser/parser.go

```diff
@@ -846,9 +847,18 @@ func listProtocols(svc *corev1.Service) map[corev1.Protocol]bool {
 func targetsForEndpoints(endpoints []util.Endpoint) []kongstate.Target {
 	targets := []kongstate.Target{}
 	for _, endpoint := range endpoints {
+		addr := endpoint.Address
+		parsed := net.ParseIP(endpoint.Address)
+		if parsed != nil {
```

Comment:
can we use the short circuit logic here, like 
```golang
if parsed == nil {
    continue
}
```

Reply:
I think we should keep these and preserve the existing behavior where the value is inserted into the target as-is, if only for it to have an obvious effect in the resulting Kong config.

We essentially trust that Kubernetes will never give us anything other than an IP address as the Endpoint address. If that _does_ somehow happen, something bizarre has happened that we don't actually know how to handle properly. Dropping it silently may or may not be the correct course of action.

Inserting it as-is into the target probably isn't correct either, but it'll at least be obvious and alert us that we need to change something.

Reply:
On further review, we construct our own non-Kubernetes Endpoint type, and that may contain the hostname from an ExternalName Service. That is valid for a target, but won't parse into an IP. Previous comment still holds if we get something that's neither an IP nor hostname; we've probably done something wrong earlier.